client/src/components/AddItem.js

```diff
@@ -54,9 +55,40 @@ const useStyles = makeStyles((theme) => ({
 
 const demoListsArray = ["Clothes", "Furniture", "Luxury"];
 
+const getItem = async (input) => {
+  const response = await fetch("/scrape", {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify({ URL: input }),
+  });
+  return response.json();
+};
+
+const emptyItem = {
+  shopURL: null,
+  title: null,
+  oldPrice: null,
+  price: null,
+  imgURL: null,
+};
+
 const AddItem = () => {
-  const [selectedItem, setSelectedItem] = useState("none");
+  const [inputLink, setInputLink] = useState("");
+  const [item, setItem] = useState(emptyItem);
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [selectedList, setSelectedList] = useState("none");
   const classes = useStyles();
+  const openPopup = () => setPopupOpen(true);
+  const addButtonClick = async () => {
+    // regex check inputLink here
+    if (inputLink.length > 0) {
+      openPopup();
+      const newItem = await getItem(inputLink);
+      setItem(newItem);
+    }
+  };
+
+  useEffect(() => setItem(emptyItem), [inputLink]);
```

Comment:
If someone happens to by typing in a link rather than paste it, this could get called way too many times. Would it make more sense to reset item when the popup is closed?

Reply:
Yeah that would make more sense. The only downside would be if the link stayed the same and the popup reopened it would send the same request, but I think that's preferable to state updates on every keystroke. I'll add the reset item on close popup functionality.